### ISSUE TITLE
mne_bids.BIDSPath.fpath error

### ISSUE THREAD

**Issue by alexrockhill** [mne-tools/mne-bids#539]
```
import mne
import mne_bids
import os.path as op
import numpy as np
out_dir = mne.utils._TempDir()
raw = mne.io.RawArray(np.random.random((1, 50000)), mne.create_info(['Ch1'], 500, ['eeg']))
raw.info['line_freq'] = 60
raw.save(op.join(out_dir, 'test-raw.fif'))
raw = mne.io.read_raw_fif(op.join(out_dir, 'test-raw.fif'))
base_path = mne_bids.BIDSPath(subject='1', root=op.join(out_dir, 'bids_dir'))
mne_bids.write_raw_bids(raw, base_path)
base_path.fdata
```

Gets the cryptic error 

```
RuntimeError: Found more than one matching data file for the requested recording. Cannot proceed due to the ambiguity. This is likely a problem with your BIDS dataset. Please run the BIDS validator on your data.
```

This does pass the validator and there was only one file saved so it probably shouldn't say more than one matching file found.

**Comment by hoechenberger** [comment]
> This does pass the validator and there was only one file saved so it probably shouldn't say more than one matching file found.

Yes, the error message is misleading here.

May I ask what output you actually expected? It's not really clear to me from your MWE. Thanks!

**Comment by adam2392** [comment]
Perhaps we could use `print_dir_tree` even in the error message here. I agree error message is noninformative here.

**Comment by alexrockhill** [comment]
I was trying to get the directory for the BIDSPath. I'd like to use it to save a beh file as by joining `bids_dir`, `path.subsesdir` and `path.basename` + `beh.tsv|json`. I could make a PR when I have a bit of time later today

**Comment by hoechenberger** [comment]
So what you would want is essentiall

> I was trying to get the directory for the BIDSPath. I'd like to use it to save a beh file as by joining `bids_dir`, `path.subsesdir` and `path.basename` + `beh.tsv|json`. I could make a PR when I have a bit of time later today

So you want the relative path **inside** the `bids_root`?

Also please note that `BIDSPath.basename` contains the extension already

**Comment by alexrockhill** [comment]
Exactly!

**Comment by hoechenberger** [comment]
I imagine you should be able to do:
```python
fpath = bids_path.copy().update(root='.').fpath
rel_path = fpath.parent if fpath.is_file() else fpath
```

**Comment by hoechenberger** [comment]
@adam2392 Why does `fpath` throw an error anyway if there's not enough info to construct a filename, but enough info to build a directory path? Shouldn't both cases work? Or is it because we're using some clever heuristics somewhere, again :)

**Comment by agramfort** [comment]
> So you want the relative path inside the bids_root?

then what do you do with it? I would update bids_root to write
anywhere with proper relative path

**Comment by adam2392** [comment]
> @adam2392 Why does `fpath` throw an error anyway if there's not enough info to construct a filename, but enough info to build a directory path? Shouldn't both cases work? Or is it because we're using some clever heuristics somewhere, again :)

So there are probably some heuristics that could get cleaned up. Namely, if extension is not provided, it will try to infer that (which I think is a fair heuristic). Rn also if suffix is not provided and if there is only one `datatype`, then it will also infer that, which could cause a breakage. Not 100% sure cuz I would need to rerun the above code that fails.


> ```
> base_path = mne_bids.BIDSPath(subject='1', root=op.join(out_dir, 'bids_dir'))
> mne_bids.write_raw_bids(raw, base_path)
> base_path.fdata
> ```

@alexrockhill  jsut to clarify, Is the desired output here `sub-1_eeg.vhdr`?

**Comment by alexrockhill** [comment]
Yes, I think with the full file path though. I really wanted something a bit different, namely `op.dirname(bids_path.fpath)`, which might be added as a field as well...

**Comment by adam2392** [comment]
> Yes, I think with the full file path though. I really wanted something a bit different, namely `op.dirname(bids_path.fpath)`, which might be added as a field as well...

For the dirname (i.e. parent), @hoechenberger suggested here https://github.com/mne-tools/mne-bids/issues/536 that we implement that as a property, which I think everyone seemed to be in favor for. That should address your `op.dirname(...)` issue.

With regards to this specific error in the Issue description, I believe the issue is that based on the bids_path you defined, there is no determination of the file path unless you also pass in `suffix` and `extension`. The reason is that there is no file path that you can run with just the `subject='1'`. This is the output of `print_dir_tree`:

```
|tmp_mne_tempdir_6klcgpad/
|--- test-raw.fif
|--- bids_dir/
|------ README
|------ dataset_description.json
|------ participants.json
|------ participants.tsv
|------ sub-1/
|--------- sub-1_scans.tsv
|--------- eeg/
|------------ sub-1_eeg.vmrk
|------------ sub-1_eeg.vhdr
|------------ sub-1_channels.tsv
|------------ sub-1_eeg.eeg
|------------ sub-1_eeg.json
```

I agree though that now if you want the `<bids root>/sub-1` dir path, that should be somehow gotten...

**Comment by alexrockhill** [comment]
I wrote a PR to update the `bids_path` in place for `write_raw_bids` that would add the suffix and extension which I think would be a nice solution

**Comment by alexrockhill** [comment]
Ok closed via #542 since the updated path is returned and I added that if `suffix is None` it'll look for all suffixes.
